src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,232 @@
+
+
+## Now, let’s start!
+
+# Step 0. Setup the environment
+
+Make sure that you installed Python 3 and the following packages:
+
+- qiskit 1.0+
+- matplotlib
+- tdqm
+
+We will use a Python package named <a href="https://github.com/vutuanhai237/qoop">$\langle \mathrm{qo}|\mathrm{op} \rangle$</a>, which is a core package for developing quantum circuit optimization frameworks. You can download it via `git` as:
+
+```
+git clone https://github.com/vutuanhai237/qoop.git
+```
+
+Note that you should put it on the same level as your Python/Jupyter Notebook file. The best is:
+
+```
+folder_name
+│   your_python_file.py
+|   your_jupyter_file.ipynb
+└───qoop
+│   └───evolution
+│       │   environment.py
+│       │   cross_over.py
+│       │   ...
+```
+
+# Step 1. Define problem (W state preparation)
+
+<figure>
+    <img src='/images/ga-qas/ga-qas_scheme.png' width="600" style="margin: auto" alt='' />
+    <figcaption>
+    Fig 1. (a) State preparation scheme based on quantum compilation technique, where U, V are ansatz and state that need to prepare[1][2]. (b) Some default ansatz, such as hyper-graph.
+    </figcaption>
+</figure>
+
+First, we define the problem, take example quantum state preparation, such as W state preparation. A 3-qubit W state is defined as:
+
+$$
+|W_3\rangle = V|000\rangle = \frac{1}{\sqrt{3}}(|001\rangle + |010\rangle + |100\rangle)
+$$
+
+If we want to prepare such state, we will need an unitary operator $V$, but in some cases, $V$ requires large resources (such as large depth and number of gates). Then, we want to find another unitary $U(\theta)$ that is approximate to $V$, $U(\theta)$ requires fewer resources than $V$ but consumes an initial to optimize $\theta$. 
+
+The problem comes from here: the structure of $U$, or ansatz $U$, affects how "hard" and "large" this optimization process is. Then, we need to find the "best" $U$ automatically. This is known as the "quantum architecture search" problem. 
+
+<!-- You can try to prepare 3-qubit W state by using $\langle qo|op \rangle$, a wiki can be found [here](https://github.com/vutuanhai237/qoop/wiki/Advance:-Custom-state-preparation): -->
+
+<!-- ```py
+from qoop.compilation.qsp import QuantumStatePreparation
+from qoop.core import ansatz, state
+
+compiler = QuantumStatePreparation(
+    u = ansatz.g2(num_qubits = 3, num_layers = 1),
+    target_state = state.w(num_qubits = 3).inverse()
+).fit(num_steps = 100) # Define the optimization process and begin to optimize in 100 iterations
+compiler.plot() # Plot optimization process
+``` -->
```

Comment:
Why was this code block commented out? If we don't want it in the blog post, it should be removed, but it seems insightful.

Reply:
Because it's a duplicate with the code below, so I want to remove it.